var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -366,71 +366,84 @@ def install(self, spec, prefix):
         else:
             options.append('--with-clanguage=C')
 
-        # PETSc depends on scalapack when '+mumps' (see depends())
-        # help PETSc pick up Scalapack from MKL
-        if spec.satisfies('+mumps'):
-            scalapack = spec['scalapack'].libs
-            options.extend([
-                '--with-scalapack-lib=%s' % scalapack.joined(),
-                '--with-scalapack=1'
-            ])
-        else:
-            options.extend([
-                '--with-scalapack=0'
-            ])
-
         # Activates library support if needed (i.e. direct dependency)
-        if '^libjpeg-turbo' in spec:
-            jpeg_library = ('libjpeg-turbo', 'libjpeg')
-        else:
-            jpeg_library = 'libjpeg'
+        jpeg_sp = spec['jpeg'].name if 'jpeg' in spec else 'jpeg'
+        scalapack_sp = spec['scalapack'].name if 'scalapack' in spec else 'scalapack'
```

Comment:
Ideally this could use `spec['package-name'].libs` (that provides an object: you can invoke `.names` to get the library names and `.directories` to get the library directories i.e. what you would `-L`).

That's not essential for this PR, but let me know if you want more info on how to do that.

Reply:
Thanks! In this case - I need `spec['scalapack'].name` so that this later code would work:
```
             library_requested = spacklibname.split(':')[0] in spec.dependencies_dict()
```
And I think it was coded this way - so that only direct dependencies are queried. [and then specified to petsc configure]

Reply:
> `spec['scalapack'].name` is evaluated to 'fujitsu-ssl2', and `spec['fujistsu-ssl2'].libs` causes an error in a subsequent statement.

I agree this should use "scalapack" for `spacklibname` (this will also work for providers of `scalapack` that just implement `.libs` and not `.scalapack_libs`).